core/src/main/java/com/amplitude/core/utilities/LoggerExtensions.kt

```diff
@@ -0,0 +1,12 @@
+package com.amplitude.core.utilities
+
+import com.amplitude.common.Logger
+
+internal fun Exception.logWithStackTrace(logger: Logger, message: String) {
+    this.message?.let {
+        logger.error("$message: $it")
+    }
+    this.stackTrace?.let {
+        logger.error("Stack trace: ${this.stackTraceToString()}")
```

Comment:
looks great! nit: it might be better to add some Amplitude prefix. Though I believe we could always leave this for later.

Reply:
Yea, I think if we want to do that, we should do that consistently across all our logs. Will leave this as is for now.